mesonbuild/scripts/run_tool.py

```diff
@@ -3,17 +3,86 @@
 
 from __future__ import annotations
 
-import itertools
+import asyncio
+import asyncio.subprocess
 import fnmatch
-import concurrent.futures
+import itertools
+import signal
+import sys
 from pathlib import Path
 
+from .. import mlog
 from ..compilers import lang_suffixes
 from ..mesonlib import quiet_git
+from ..mtest import sh_quote, determine_worker_count, complete_all
 import typing as T
 
-if T.TYPE_CHECKING:
-    import subprocess
+Info = T.TypeVar("Info")
+
+async def run_with_buffered_output(cmdlist: T.List[str]) -> int:
+    """Run the command in cmdlist, buffering the output so that it is
+       not mixed for multiple child processes.  Kill the child on
+       cancellation."""
+    quoted_cmdline = ' '.join(sh_quote(x) for x in cmdlist)
```

Comment:
What is quoted_cmdline, and why does mtest have a function that isn't mesonlib.join_args for this?

Reply:
In mtest it's there because it also includes the environment settings, which is not really possible to write without Unix syntax afaik. Will use join_args here.

Reply:
quoted_cmdline is only used for printing on stdout btw.